pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -456,6 +458,75 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	if cd.Spec.MachineManagementStrategy != nil && cd.Spec.MachineManagementStrategy.Strategy == "Central" {
+		targetNamespace := cd.Status.TargetNamespace
+		if targetNamespace == "" {
+			targetNamespace = apihelpers.GetResourceName(cd.Name+"-targetns", utilrand.String(5))
+		}
+
+		cdLog.Info("Creating the target namespace ", targetNamespace)
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: targetNamespace},
+		}
+		if err := r.Create(context.TODO(), ns); err != nil && !apierrors.IsAlreadyExists(err) {
+			return reconcile.Result{}, fmt.Errorf("failed to create target namespace %q: %w", ns.Name, err)
+		}
+
+		if cd.Status.TargetNamespace == "" {
+			cd.Status.TargetNamespace = targetNamespace
+			if err := r.Status().Update(context.TODO(), cd); err != nil {
+				cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not set cluster target namespace")
+				return reconcile.Result{Requeue: true}, nil
+			}
+		}
```

Comment:
I think this block needs to go before line 467, otherwise it's possible we'd leak a namespace we created, but failed to record, and then lost the relationship.